appmap/test/test_recording.py

```diff
@@ -0,0 +1,64 @@
+"""Test recording context manager"""
+import json
+import os
+import platform
+import sys
+from importlib import reload
+
+import pytest
+
+FIXTURE_DIR = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)),
+    'data'
+    )
+
+
+@pytest.mark.datafiles(
+    os.path.join(FIXTURE_DIR, 'appmap.yml'),
+    os.path.join(FIXTURE_DIR, 'example_class.py'),
+    os.path.join(FIXTURE_DIR, 'expected.appmap.json')
+    )
+def test_recording_works(monkeypatch, datafiles):
+    with open(datafiles / 'expected.appmap.json') as f:
```

Comment:
```suggestion
    with open(os.path.join(datafiles, 'expected.appmap.json')) as f:
```

Reply:
I wrote it this way, because the [pytest-datafiles](https://pypi.org/project/pytest-datafiles/) doc uses the `/` operator. Happy to change it, but what's the advantage to using `os.path.join`?

Reply:
it is useful only when on Windows which uses \ instead of /. But I guess they way Windows is moving towards linux base, might not be an issue long term.

Reply:
Rereading the doc, `datafiles` is a `py.path`. So, `datafiles / 'expected.appmap.json'` uses `py.path.__div__` to construct a path with the correct separator. (If they were strings, trying to use the division operator would raise an exception, right?)

But, having said that, I see that my copying and pasting from the pytest-datafiles doc has left us with a mix of `/` and `os.path.join`, with more uses of `os.path.join` than `/`. I'll switch to `os.path.join` to make them consistent.